phys2bids/physio_obj.py

```diff
@@ -594,10 +592,25 @@ def auto_trigger_selection(self):
                                 'Please run phys2bids specifying the -chtrig argument.')
             else:
                 self.trigger_idx = indexes[0]
-                LGR.info(f'{self.ch_name[self.trigger_idx]} selected as trigger channel')
         else:
-            raise Exception('No trigger channel automatically found. Please run phys2bids '
-                            'specifying the -chtrig argument.')
+            # Time-domain automatic trigger detection
+            # Initialize distance array
+            mean_d = [np.nan] * (self.ch_amount - 1)
+            # Loop through channels
+            for n in range(1, self.ch_amount):
+                # Get timeseries
+                s = self.timeseries[n]
+                # Normalize to [0,1]
+                s = (s - min(s)) / (max(s) - min(s))
+                # Calculate distance to the closest signal limit (min or max)
+                d = np.minimum(abs(s - max(s)), abs(s - min(s)))
+                # Store the mean distance
+                mean_d[n - 1] = np.mean(d)
```

Comment:
I think this should probably be done by calculating the mean, min and max along a given axis with numpy. I think avoiding for loops makes the code easier to read and less prone to errors.

Reply:
Also, I think the variables should be given a name that is more representative of what the values they hold, e.g. `channel_ts` and `distance` instead of `s` and `d`.